source/application/DbgHelp/DbgHelpStackWalkCallback.cpp

```diff
@@ -484,70 +463,89 @@ DbgHelpStackWalkCallback::GetParameterValue(const ParamInfo& param_info)
-                return char32_t{};
-            case DbgHelpBasicType::BasicType::Int8:
-                return int8_t{};
-            case DbgHelpBasicType::BasicType::UnsignedInt8:
-                return uint8_t{};
-            case DbgHelpBasicType::BasicType::Int16:
-                return int16_t{};
-            case DbgHelpBasicType::BasicType::UnsignedInt16:
-                return uint16_t{};
-            case DbgHelpBasicType::BasicType::Int32:
-                return int32_t{};
-            case DbgHelpBasicType::BasicType::UnsignedInt32:
-                return uint32_t{};
-            case DbgHelpBasicType::BasicType::Int64:
-                return int64_t{};
-            case DbgHelpBasicType::BasicType::UnsignedInt64:
-                return uint64_t{};
-            case DbgHelpBasicType::BasicType::Float:
-                return float{};
-            case DbgHelpBasicType::BasicType::Double:
-                return double{};
             }
-        }
-
-        std::optional<IDbgHelpStackWalkCallback::ParameterValue> operator()(DbgHelpPointerType t)
-        {
-            return static_cast<void*>(nullptr);
-        }
+            else if constexpr (std::is_same_v<T, DbgHelpBasicType>)
+            {
+                switch (type.m_type)
+                {
+                case DbgHelpBasicType::BasicType::Void:
+                    return std::nullopt;
+                case DbgHelpBasicType::BasicType::Char:
+                    return char{};
+                case DbgHelpBasicType::BasicType::WideChar:
+                    return wchar_t{};
+                case DbgHelpBasicType::BasicType::Char16:
+                    return char16_t{};
+                case DbgHelpBasicType::BasicType::Char32:
+                    return char32_t{};
+                case DbgHelpBasicType::BasicType::Int8:
+                    return int8_t{};
+                case DbgHelpBasicType::BasicType::UnsignedInt8:
+                    return uint8_t{};
+                case DbgHelpBasicType::BasicType::Int16:
+                    return int16_t{};
+                case DbgHelpBasicType::BasicType::UnsignedInt16:
+                    return uint16_t{};
+                case DbgHelpBasicType::BasicType::Int32:
+                    return int32_t{};
+                case DbgHelpBasicType::BasicType::UnsignedInt32:
+                    return uint32_t{};
+                case DbgHelpBasicType::BasicType::Int64:
+                    return int64_t{};
+                case DbgHelpBasicType::BasicType::UnsignedInt64:
+                    return uint64_t{};
+                case DbgHelpBasicType::BasicType::Float:
+                    return float{};
+                case DbgHelpBasicType::BasicType::Double:
+                    return double{};
+                default:
+                    /*
+                     * VS is stupid and warns even if everything is covered.
+                     */
+                    assert(false);

```

Comment:
I thought about it, it could perhaps be useful if something else will get added to the enum and the person will forget to change all occurrences. But then it should be added to the other one or two places where it's used. I could do it in another PR where I intend to fix some more annoying warnings.